mmdet/datasets/custom.py

```diff
@@ -210,9 +211,20 @@ def prepare_train_img(self, idx):
         img_info = self.data_infos[idx]
         ann_info = self.get_ann_info(idx)
         results = dict(img_info=img_info, ann_info=ann_info)
+
+        copy_past_idx = randint(0, len(self) - 1)
+        while copy_past_idx == idx:
+            copy_past_idx = randint(0, self.__len__() - 1)
+        copy_past_img_info = self.data_infos[copy_past_idx]
+        copy_past_ann_info = self.get_ann_info(copy_past_idx)
+
         if self.proposals is not None:
             results['proposals'] = self.proposals[idx]
         self.pre_pipeline(results)
+
+        results['copy_paste'] = dict(img_info=copy_past_img_info, ann_info=copy_past_ann_info)
+        self.pre_pipeline(results['copy_paste'])
```

Comment:
don't we need to define some parameter enabling/disabling preliminary actions for copy_paste augmentation (if it is not used)?

Reply:
Probably, yes. Check for the `CopyPaste` augmentation added.